homeassistant/components/shelly/entity.py

```diff
@@ -238,3 +275,98 @@ def device_state_attributes(self):
             return None
 
         return self.description.device_state_attributes(self.block)
+
+
+class ShellyRestAttributeEntity(entity.Entity):
```

Comment:
Why are you not multi-inheriting from the DataCoordinatorEntity ? That would allow you to drop `async_added_to_hass`, `should_poll` and `_update_callback`